src/vse_sync_pp/analyzers/analyzer.py

```diff
@@ -265,3 +270,193 @@ def explain(self, data):
+        # samples in the initial transient period are ignored
+        self._transient = config.parameter('transient-period/s')
+        # minimum test duration for a valid test
+        self._duration = config.parameter('min-test-duration/s')
+        # limit initial tau observation windows from 1 to 9k taus
+        taus_below_tenkey = np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 20, 30, 40, 50, 60, 70, 80, 90,
+                                     100, 200, 300, 400, 500, 600, 700, 800, 900, 1000, 2000, 3000,
+                                     4000, 5000, 6000, 7000, 8000, 10000])
+        # observation window upper bound limited to 100k samples in 5k increments
+        taus_above_tenkey = np.arange(10001, 100000, 5000)
+        # `taus_list` contains range limit of obervation window intervals for which to compute the statistic
+        self._taus_list = np.concatenate((taus_below_tenkey, taus_above_tenkey))
+        self._rate = None
+        self._lpf_signal = None
+        # atributtes to be initialized in derived class
+        self._accuracy = None
+
+    def calculate_limit(self, tau):
+        # return limit based on `tau`
+        if self._accuracy is None:
+            raise AttributeError('atributte not initialized')
+        for (low, high), f in self._accuracy.items():
+            if ((low is None or tau > low) and (tau <= high)):
+                return f(tau) * (self._limit / 100)
+
+    def out_of_range(self, taus, samples):
+        # return true if any value in `tdevs` is out of range
+        for tau, sample in zip(taus, samples):
+            mask = self.calculate_limit(tau)
+            if mask <= sample:
+                return True
+        return False
+
+    def prepare(self, rows):
+        idx = 0
+        try:
+            tstart = rows[0].timestamp + self._transient
+        except IndexError:
+            pass
+        else:
+            while idx < len(rows):
+                if tstart <= rows[idx].timestamp:
+                    break
+                idx += 1
+        return super().prepare(rows[idx:])
+
+    def calculate_filter(self, input_signal):
+        """Calculate digital low-pass filter from `input_signal`
+
+        scipy_signal.butter input arguments:
+          order of the filter: 1
+          critical Frequency:
+            cutoff Frequency: 0.1Hz
+            for digital filters is normalized from 0 to 1 where 1 is Nyquist freqency.
+          btype is band type or type of filter
+          analog=False since it is always a digital filter
+        scipy_signal.butter return arguments:
+          `numerator` coefficient vector and `denominator coefficient vector of the butterworth digital filter
+        """
+        numerator, denominator = scipy_signal.butter(1, 0.1 / (1 / 2), btype="low", analog=False, output="ba")
+        lpf_signal = scipy_signal.filtfilt(numerator, denominator, input_signal.terror)
+        lpf_signal = lpf_signal[self._transient:len(lpf_signal)]
```

Comment:
`self._transient` is the only use of `self` in this method. Make it a regular function? and just return `lpf_signal`? (i.e. the return value of `scipy_signal.filtfilt`) This will allow the caller to control what `input_signal` is and which parts of the output signal to discard, a clean division of responsibility.